uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Reminder.php

```diff
@@ -1009,4 +993,77 @@ private static function getUfMatchContact($contact_id) {
     return $uf_match_data;
   }
 
+  /**
+   * Constructs a query condition that excludes "completed" activities.
+   *
+   * @return string
+   */
+  private static function getCompletedActivitiesExclusionQuery() {
+    $documentIncompleteStatuses = implode(',', self::_getDocumentIncompleteStatuses());
+    $documentTypesIds = implode(',', self::_getTypesIdsForComponent('CiviDocument'));
+    $taskIncompleteStatuses = implode(',', self::_getTaskIncompleteStatuses());
+    $taskTypesIds = implode(',', self::_getTypesIdsForComponent('CiviTask'));
+
+    return "(
+      (
+        a.status_id IN ($documentIncompleteStatuses)
+        AND a.activity_type_id IN ($documentTypesIds)
+      )
+      OR
+      (
+        a.status_id IN ($taskIncompleteStatuses)
+        AND a.activity_type_id IN ($taskTypesIds)
+      )
+    )";
+  }
+
+  /**
+   * Returns incomplete statuses for Documents
+   *
+   * @return array
+   */
+  private static function _getDocumentIncompleteStatuses() {
+    return [
+      CRM_Tasksassignments_BAO_Document::STATUS_AWAITING_UPLOAD,
+      CRM_Tasksassignments_BAO_Document::STATUS_AWAITING_APPROVAL
+    ];
+  }
+
+  /**
+   * Returns Activity Types IDs for a given component name,
+   * for example, "CiviTask" or "CiviDocument"
+   *
+   * @param string $componentName eg
+   * @return array
+   */
+  private static function _getTypesIdsForComponent($componentName) {
+    $taskTypesIds = [];
+    $taskTypes = ActivityService::findByComponent($componentName);
+
+    foreach ($taskTypes as $taskType) {
+      $taskTypesIds[] = $taskType['value'];
+    }
+
+    return $taskTypesIds;
+  }
+
+  /**
+   * Returns incomplete statuses for Tasks
+   *
+   * @return array
+   */
+  private static function _getTaskIncompleteStatuses() {
```

Comment:
No need to start function name with underscore here also.

Reply:
Done.